ng_package/angular_package_format.bzl

```diff
@@ -2,56 +2,427 @@ load("//ng_project/config:compilation_mode.bzl", "partial_compilation_transition
+            "dtsBundleRelativePath": m.dts_bundle_relative_path,
+            # If the paths for that entry-point were guessed (e.g. "ts_library" rule or
+            # "ng_module" without flat module bundle), we pass this information to the packager.
+            "guessedPaths": m.guessed_paths,
+        }
+
+    for ep in ctx.attr.side_effect_entry_points:
+        if not metadata_arg[ep]:
+            known_entry_points = ",".join([e.module_name for e in collected_entry_points])
+            fail("Unknown entry-point (%s) specified to include side effects. " % ep +
+                 "The following entry-points are known: %s" % known_entry_points)
+
+    fesm_rollup_config = _write_rollup_config(
+        ctx,
+        ctx.bin_dir.path,
+        metadata_arg,
+        ctx.attr.side_effect_entry_points,
+        dts_mode = False,
+    )
+    fesm_rollup_inputs = depset(static_files, transitive = [unscoped_all_entry_point_esm2022_depset])
+    fesm_bundles_out = _run_rollup(ctx, fesm_rollup_config, fesm_rollup_inputs, dts_mode = False)
+
+    dts_rollup_config = _write_rollup_config(
+        ctx,
+        ctx.bin_dir.path,
+        metadata_arg,
+        ctx.attr.side_effect_entry_points,
+        dts_mode = True,
+    )
+    dts_rollup_inputs = depset(static_files, transitive = [unscoped_all_entry_point_dts_depset])
+    dts_bundles_out = _run_rollup(ctx, dts_rollup_config, dts_rollup_inputs, dts_mode = True)
+
+    packager_inputs.append(fesm_bundles_out)
+    packager_inputs.append(dts_bundles_out)
+
+    # Encodes the package metadata with all its entry-points into JSON so that
+    # it can be deserialized by the packager tool. The struct needs to match with
+    # the `PackageMetadata` interface in the packager tool.
+    packager_args.add(json.encode(struct(
+        npmPackageName = npm_package_name,
+        entryPoints = metadata_arg,
+        fesmBundlesOut = _serialize_file(fesm_bundles_out),
+        dtsBundlesOut = _serialize_file(dts_bundles_out),
+    )))
+
+    if ctx.file.readme_md:
+        packager_inputs.append(ctx.file.readme_md)
+        packager_args.add(ctx.file.readme_md.path)
+    else:
+        # placeholder
+        packager_args.add("")
+
+    if ctx.file.license:
+        packager_inputs.append(ctx.file.license)
+        packager_args.add(ctx.file.license.path)
+    else:
+        #placeholder
+        packager_args.add("")
+
+    packager_args.add(_serialize_files_for_arg(esm2022))
+    packager_args.add(_serialize_files_for_arg(static_files))
+
+    packager_args.add(json.encode(ctx.attr.side_effect_entry_points))
+
+    ctx.actions.run(
+        progress_message = "Angular Packaging: building npm package %s" % str(ctx.label),
+        mnemonic = "AngularPackage",
+        inputs = packager_inputs,
+        outputs = [npm_package_directory],
+        executable = ctx.executable.ng_packager,
+        arguments = [packager_args],
+        env = {
+            "BAZEL_BINDIR": ".",
```

Comment:
OOC: can these be `ctx.bin_dir.path` or whatever the official variable is? or did the error actually recommend using `.`? (I feel like I have something in mind)

Reply:
The error when there is no `BAZEL_BINDIR` defined recommends using `.`.

I actually tried using `ctx.bin_dir.path` however, that puts us in `bazel-out/k8-fastbuild/bin`, but since we need to load files from there as well as from locations from the transitioned code as well which ends up in a different subfolder.

Reply:
Interesting. Would love to keep an eye on this. I'd expect `bin_dir` to be the transitioned code root, with all the stuff from `k8-fastbuild` also being in there.